helpers/clean_pnp_backdata.py

```diff
@@ -0,0 +1,320 @@
+                              'q0216': '222',
+                              'q0218': '223',
+                              'q0222': '211',
+                              'q0224': '205',
+                              'q0226': '205',
+                              'q0228': '206',
+                              'q0230': '605',
+                              'q0301': '212',
+                              'q0303': '214',
+                              'q0305': '216',
+                              'q0307': '242',
+                              'q0309': '243',
+                              'q0311': '244',
+                              'q0313': '245',
+                              'q0315': '246',
+                              'q0317': '247',
+                              'q0321': '248',
+                              'q0323': '249',
+                              'q0325': '218',
+                              'q0329': '250',
+                              'q0501': '501',
+                              'q0502': '502',
+                              'q0503': '503',
+                              'q0504': '504',
+                              'q0505': '505',
+                              'q0506': '506',
+                              'q0507': '507',
+                              'q0508': '508',
+                              'q0509': '405',
+                              'q0511': '407',
+                              'q0513': '409',
+                              'q0515': '411',
+                              'q0601': '601',
+                              'q0602': '602',
+                              'q0701': '708',
+                              'q0702': '712',
+                              'q0902': '302',
+                              'q0904': '303',
+                              'q0906': '304',
+                              'q0908': '305'}
+
+    # Remove unwanted columns
+    df = df.drop(columns=columns_to_remove_list)
+
+    # Rename wanted columns
+    df = df.rename(columns=columns_to_rename_dict)
+
+    # convert column datatypes
+    df = convert_column_datatypes(df)
+
+    # Populate the statusencoded column
+    df = get_status_encoded(df)
+
+    # Filter the DataFrame for rows where statusencoded is 210 or 211
+    df = df[df['statusencoded'].isin([210, 211])]
+
+    # Identify the key business
+    df = identify_key_business(df)
+
+    # Identify the osmotherly business
+    df = identify_osmotherly_businesses(df)
+
+    return df
+
+
+def main(input_file, output_file):
+    """ Main function to clean the PNP backdata.
+
+    Read in csv file as a dataframe, clean with clean_pnp_backdata function,
+    and save dataframe as a csv file.
+
+    Example cmd executiion: python clean_pnp_backdata.py input.csv output.csv
+
+    Args:
+        input_file (str): The path to the input CSV file.
+        output_file (str): The path to save the cleaned CSV file.
+    Return:
+        None
+    """

```

Comment:
I wanted to make it a bit easier for me to run this, so I updated the code locally to this, what do you think? 

`    ```
    rootpath = "R:/BERD Results System Development 2023/DAP_emulation/2021_surveys/PNP"
    df = pd.read_csv(os.path.join(rootpath, input_file))

    # Clean the DataFrame
    cleaned_df = clean_pnp_backdata(df)

    # Save the cleaned DataFrame to the output CSV file
    cleaned_df.to_csv(os.path.join(rootpath, output_file), index=False)
```
`

Reply:
Much better aye.